neo/Consensus/ConsensusContext.cs

```diff
@@ -13,34 +13,42 @@
 
 namespace Neo.Consensus
 {
-    internal class ConsensusContext : IDisposable
+    internal class ConsensusContext : IConsensusContext
     {
         public const uint Version = 0;
-        public ConsensusState State;
-        public UInt256 PrevHash;
-        public uint BlockIndex;
-        public byte ViewNumber;
-        public Snapshot Snapshot;
-        public ECPoint[] Validators;
-        public int MyIndex;
-        public uint PrimaryIndex;
-        public uint Timestamp;
-        public ulong Nonce;
-        public UInt160 NextConsensus;
-        public UInt256[] TransactionHashes;
-        public Dictionary<UInt256, Transaction> Transactions;
-        public byte[][] Signatures;
-        public byte[] ExpectedView;
-        private KeyPair KeyPair;
+        public ConsensusState State { get; set; }
+        public UInt256 PrevHash { get; set; }
+        public uint BlockIndex { get; set; }
+        public byte ViewNumber { get; set; }
+        public ECPoint[] Validators { get; set; }
+        public int MyIndex { get; set; }
+        public uint PrimaryIndex { get; set; }
+        public uint Timestamp { get; set; }
+        public ulong Nonce { get; set; }
+        public UInt160 NextConsensus { get; set; }
+        public UInt256[] TransactionHashes { get; set; }
+        public Dictionary<UInt256, Transaction> Transactions { get; set; }
+        public byte[][] Signatures { get; set; }
+        public byte[] ExpectedView { get; set; }
+        private Snapshot snapshot;
+        private KeyPair keyPair;
         private readonly Wallet wallet;
 
         public int M => Validators.Length - (Validators.Length - 1) / 3;
+        public Header PrevHeader => snapshot.GetHeader(PrevHash);
 
         public ConsensusContext(Wallet wallet)
         {
             this.wallet = wallet;
         }
 
+        public bool RejectTx(Transaction tx, bool verify)
```

Comment:
This section needs to be moved back to `ConsensusService`.

Reply:
Good idea! Now I think I can do it without breaking encapsulation.